doc/specs/stdlib_sorting.md

```diff
@@ -0,0 +1,644 @@
+
+`work` (optional): shall be a rank one array of the same type as
+array, and shall have at least `size(array)/2` elements. It is an
+`intent(inout)` argument. It is intended to be used as "scratch"
+memory for internal record keeping. If associated with an array in
+static storage, its use can significantly reduce the stack memory
+requirements for the code. Its contents on return are undefined.
+
+##### Notes
+
+`ORD_SORT` implements a hybrid sorting algorithm combining
+`merge sort`, and `insertion sort`. For most purposes it behaves like
+a `merge sort`, providing worst case `O(N Ln(N))` run time performance
+for most random arrays, that is typically slower than `SORT`.
+However, if the array has significant runs of decreasing or
+non-decreasing values, performance can be much better than
+`SORT`, with `O(N)` behavior on uniformly decreasing, or
+non-decreasing arrays. The optional `work` array replaces "scratch"
+memory that would otherwise be allocated on the stack. If `array` is of
+any type `REAL` the order of its elements on return undefined if any
+element of `array` is a `NaN`.  Sorting of `CHARACTER(*)` and
+`STRING_TYPE` arrays are based on the operator `>`, and not on the
+function `LGT`.
+
+
+##### Example
+
+```fortran
+    ...
+    ! Read arrays from sorted files
+    call read_sorted_file( 'dummy_file1', array1 )
+    call read_sorted_file( 'dummy_file2', array2 )
+    ! Concatenate the arrays
+	array = [ array1, array2 ]
+    ! Sort the resulting array
+    call ord_sort( array, work )
+    ! Process the sorted array
+    call array_search( array, values )
+    ...
+```
+
+#### `sort` - sorts an input array
+
+##### Status
+
+Experimental
+
+##### Description
+
+Returns an input array with the elements sorted in order of increasing
+value.
+
+##### Syntax
+
+`call [[stdlib_sorting(module):sort(subroutine)]]sort ( array )`
+
+##### Class
+
+Pure generic subroutine.
+
+##### Arguments
+
+`array` : shall be a rank one array of any of the types:
+`integer(int8)`, `integer(int16)`, `integer(int32)`, `integer(int64)`,
+`real(sp)`, `real(dp)`, `real(qp)`. `character(*)`, or
+`type(string_type)`. It is an `intent(inout)` argument. On return its
+input elements will be sorted in order of non-decreasing value.
+
+##### Notes
+
+`SORT` implements a hybrid sorting algorithm combining
+`quicksort`, `merge sort`, and `insertion sort`. For most purposes it
+behaves like a `quicksort` with a median of three partition, providing
+good, `O(N Ln(N))`, run time performance for most random arrays, but
+defaulting to `merge sort` if the structure of the array results in
+the `quicksort` not converging as rapidly as expected. If `array` is of
+any type `REAL`, the behavior of the sorting is undefined if any
+element of `array` is a `NaN`.  Sorting of `CHARACTER(*)` and
+`STRING_TYPE` arrays are based on the operators `<`, `<=`, `>`, and
+`>=`, and not on the functions `LLT`, `LLE`, `LGT`, or `LGE`.
+
+##### Example
+
+```fortran
```

Comment:
As for the previous example, I suggest to treplace this example by a small demo program.

Reply:
Done.